TaskSQL/Queries.sql

```diff
@@ -0,0 +1,296 @@
+	('2015', '3', '2'), 
+	('545', '2', '3'), 
+	('4125', '4', '5'), 
+	('2158', '5', '4');
+
+--Filling out the table "Cards"
+INSERT INTO Cards(AccountId, CardNumber, CardBalance) 
+values ('6', '1234567891025684', '60.21'), 
+	('5', '6589423682354', '125.03'), 
+	('4', '5454886452213', '235.06'), 
+	('3', '54848303255452', '1225'), 
+	('2', '458787933656006', '300.21'), 
+	('1', '8759232656552', '125.50'), 
+	('6', '87592326565', '200.30');
+
+--Item 1: Output the name of banks that are in the city X
+SELECT BankName 
+FROM Banks 
+	INNER JOIN BanksCities ON Banks.Id = BanksCities.BankId 
+	INNER JOIN Cities ON Cities.Id = BanksCities.CityId 
+WHERE Cities.City LIKE 'Gomel';
+
+--Item 2: Get a list of cards indicating the name of the owner, balance and the name of the bank
+SELECT CardNumber, CardBalance, bankName, OwnerName 
+FROM Cards 
+	INNER JOIN Accounts ON Cards.AccountId = Accounts.Id 
+	INNER JOIN Banks ON Accounts.BankId = Banks.Id 
+	INNER JOIN Owners ON Accounts.OwnerId = Owners.Id;
+
+--Item 3: Show a list of bank accounts whose balance does not coincide with the sum of balance on cards. In a separate column, output the difference
+SELECT DISTINCT Accounts.Id,
+	MAX(Accounts.Balance) 'AccountBalance', 
+	SUM(Cards.CardBalance) 'SumOdCardBalanceWithOneAccount', 
+	ABS(SUM(Cards.CardBalance) - MAX(Accounts.Balance)) 'Difference'
+FROM Accounts 
+	INNER JOIN Cards ON Accounts.Id = Cards.AccountId 
+GROUP BY (Accounts.Id)
+HAVING SUM(Cards.CardBalance) <> MAX(Accounts.Balance)
+
+--Item 4: Output count bank cards for everyone social status with GROUP BY
+SELECT SocialStatus, COUNT (Cards.CardNumber) 'CountCards'  
+FROM SocialStatus 
+	FULL OUTER JOIN Owners ON SocialStatus.Id = Owners.SocialStatusId
+	FULL OUTER JOIN Accounts ON Accounts.OwnerId = Owners.Id
+	FULL OUTER JOIN Cards ON Accounts.Id = Cards.AccountId
+GROUP BY SocialStatus.SocialStatus;
+
+--Item 4: Output count bank cards for everyone social status WITHOUT
+SELECT SocialStatus, 
+	(SELECT COUNT (Cards.CardNumber)
+	FROM Cards
+		FULL OUTER JOIN Accounts ON Accounts.Id = Cards.AccountId
+		FULL OUTER JOIN Owners ON Owners.Id = Accounts.OwnerId
+		WHERE Owners.SocialStatusId = SocialStatus.Id
+		) 'CountCards'
+FROM SocialStatus
+
+--Item 5: Write Stored Procedure which will add $ 10 to each bank account for a specific status social
+GO
+CREATE PROCEDURE AddTen @SocStatusId INT
+AS 
+BEGIN
+	Update Accounts SET Balance = Balance + 10 
+	FROM Accounts 
+		INNER JOIN Owners ON Accounts.OwnerId = Owners.Id 
+	WHERE SocialStatusId = @SocStatusId
+END;
+
+DECLARE @UpAccountBalanceForSocialStatusId INT = 1
+Exec AddTen @UpAccountBalanceForSocialStatusId
+
+--Item 6: Get a list of available funds for each client
+SELECT Owners.OwnerName, (SumBalances.SumBalance - SUM(Cards.CardBalance)) AS FreeMoneyOfAccount
+FROM (SELECT Owners.Id, SUM (Accounts.Balance) 'SumBalance'
+	  FROM Accounts 
+		  FULL OUTER JOIN Owners ON Owners.Id = Accounts.OwnerId
+	  GROUP BY (Owners.Id)) 

```

Comment:
You can create a query for this task without subQuery. In main query you already have joined Accounts and Owners. And in the main query, you already group by Owners.OwnerName field (here should be group by ID instead)